osu.Game/Rulesets/Scoring/HitResult.cs

```diff
@@ -122,19 +122,33 @@ public enum HitResult
     public static class HitResultExtensions
     {
         /// <summary>
-        /// Whether a <see cref="HitResult"/> increases/decreases the combo, and affects the combo portion of the score.
+        /// Whether a <see cref="HitResult"/> increases the combo.
         /// </summary>
-        public static bool AffectsCombo(this HitResult result)
+        public static bool IncreasesCombo(this HitResult result)
         {
             switch (result)
             {
-                case HitResult.Miss:
                 case HitResult.Meh:
                 case HitResult.Ok:
                 case HitResult.Good:
                 case HitResult.Great:
                 case HitResult.Perfect:
                 case HitResult.LargeTickHit:
+                    return true;
+
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Whether a <see cref="HitResult"/> breaks the combo and resets it back to zero.
+        /// </summary>
+        public static bool BreaksCombo(this HitResult result)
+        {
+            switch (result)
```

Comment:
Not sure why this was reimplemented manually instead of using `HitResult.IsHit()`. Oh well.

Reply:
`IsHit` has hit results that may not affect combo (such as `SmallTickMiss` and `IgnoreMiss`).

Reply:
`AffectsCombo() && IsHit()`. Or with the latter negated in the converse case.